### PR TITLE
CCORE-639: Create new AppCenter DLL client package for Showroom code assets

### PR DESCRIPTION
**Pull request by ryanbrookfield**
### Purpose of this PR

Add in Showroom specific packages to AppCenter

### PR Merge order

- [ ] PR https://github.com/Unity-Technologies/createstudio_core/pull/433
- This PR

### Testing status

Tested by having CS core look at this branch to push libs to AppCenter.
Verified that the uploaded Artifacts contained the proper *.dll files and were named correctly and had the correct distro list

This would have triggered all core tests

### Technical risk

Minimal.
This is only adding an additional upload to AppCenter based on preexisting code.

### CODE REVIEW

File: vars/cloudBuildPipeline.groovy [hunk 1]
```diff
@@ -487,20 +489,36 @@ def call(body) {
...
                                     }
                                 }
-                                project = sh(returnStdout: true, script: "find . -maxdepth 1 -type d | grep ${libDir} | sed -e 's/\\.\\///g'").trim()
+                                // Upload Showroom specific Libraries to AppCenter
+                                project = sh(returnStdout: true, script: "find . -maxdepth 1 -type d | grep ${showroomlibDir} | sed -e 's/\\.\\///g'").trim()
                                 sh('apk update && apk add git')
                                 sh("ls -la ${project}")
                                 echo ("Built ${project} !")
                                 ZipAndArchive(project: "${project}")
+                                def ShowroomAppCenterURL = UploadToAppCenter(version: 'showroom-0.1.0', projectType: "${PROJECT_TYPE}", projectPath: "${project}.zip", distGroups: 'ShowroomDevelopers')
```
Comment (jrowinski3d) [inline vars/cloudBuildPipeline.groovy:514]: whats the version here for and any reason its hard coded?

**Review by bdebrincunity — DISMISSED** [review]

**Review by jrowinski3d — CHANGES_REQUESTED** [review]

File: vars/cloudBuildPipeline.groovy (hunk 1 above)
Comment (ryanbrookfield) [reply vars/cloudBuildPipeline.groovy:514]: ahhhh, I see. Need to change that to something like `showroom-${envVars.VERSION}`

Comment (jrowinski3d) [reply vars/cloudBuildPipeline.groovy:514]: Is this what will be reflected in the `release` name?

Comment (ryanbrookfield) [reply vars/cloudBuildPipeline.groovy:514]: yes, so the appcenter package will look like the below based on UploadToAppCenter coding.

[String fullBuildVersion = "${BRANCH_NAME}-${buildVer}"](https://github.com/Unity-Technologies/createstudio_jenkins_lib/blob/develop/vars/UploadToAppCenter.groovy#L44)

where buildVer = version

Comment (jrowinski3d) [reply vars/cloudBuildPipeline.groovy:514]: perfect, let me know how it goes!

Comment (AnthonyV44) [reply vars/cloudBuildPipeline.groovy:514]: @lairdm do you have a preference on the names of these AppCenter packages?

Comment (ryanbrookfield) [reply vars/cloudBuildPipeline.groovy:514]: chatted around and looks like we are going to go with `${BRANCH_NAME}-showroom-${VERSION}`. Ill make the change and test. Packages should be visible tomorrow

File: vars/UploadToAppCenter.groovy [hunk 2]
```diff
@@ -41,7 +42,7 @@ String call(Map args = [:]) {
     if (DEBUG) { echo "DEBUG: appName: ${appName} -- Last Commit Message: ${lastCommit}" }
 
     // Grab VERSION var, otherwise default to 0.1.0
-    String fullBuildVersion = "${BRANCH_NAME}-${buildVer}"
+    String fullBuildVersion = ["${BRANCH_NAME}","${mergedArgs.projectName}","${buildVer}"].findAll({it != ''}).join('-')
```
Comment (ryanbrookfield) [inline vars/UploadToAppCenter.groovy:45]: Changed the logic here for the version string to be a bit more dynamic if you want to add in the Project name.

AppCenter will look like this:

Screen Shot 2021-01-07 at 10 29 56 AM

**Review by jrowinski3d — APPROVED** [review]

**Review by bdebrincunity — APPROVED** [review]

**Pull request merged**